python/ray/tune/tune.py

```diff
@@ -83,26 +113,56 @@ def run_experiments(experiments=None,
         List of Trial objects, holding data for each executed trial.
 
     """
-
-    if scheduler is None:
-        scheduler = FIFOScheduler()
-
-    if search_alg is None:
-        search_alg = BasicVariantGenerator()
-
-    search_alg.add_configurations(experiments)
-
-    runner = TrialRunner(
-        search_alg,
-        scheduler=scheduler,
-        launch_web_server=with_server,
-        server_port=server_port,
-        verbose=verbose,
-        queue_trials=queue_trials,
-        trial_executor=trial_executor)
+    # This is important to do this here
+    # because it schematize the experiments
+    # and it conducts the implicit registration.
+    experiments = convert_to_experiment_list(experiments)
+    checkpoint_dir = _find_checkpoint_dir(experiments)
+    if checkpoint_dir:
+        logger.info("Using checkpoint dir: {}.".format(checkpoint_dir))
+    runner = None
+
+    if resume:
```

Comment:
IMO the control flow here is kind of obfuscated. There should be only one branch that calls into try_restore_runner. All the prompting and flags logic can be done before that.